src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt

```diff
@@ -31,22 +31,26 @@ class InvalidBookable : RuntimeException("Bookable does not exist")
 @RestController
 @RequestMapping("/v1/location/{locationId}/bookable")
 @Transactional
-class BookableController(private val bookableRepository: BookableRepository, private val locationRepository: LocationRepository, val bookingRepository: BookingRepository) {
+class BookableController(private val bookableRepository: BookableRepository, val bookingRepository: BookingRepository) {
     /**
      * Get a bookable
      */
     @GetMapping("/{bookableId}")
-    fun getBookable(@PathVariable("locationId") location: String, @PathVariable("bookableId") bookable: String): BookableResource {
-        locationRepository.getLocations().find { it.id == location } ?: throw LocationNotFound()
-        return BookableResource(bookableRepository.getAllBookables().find { it.id == bookable } ?: throw BookableNotFound())
+    fun getBookable(@PathVariable("locationId") location: Location?, @PathVariable("bookableId") bookable: Bookable?): BookableResource {
```

Comment:
spring will automatically use the locationRepo to map locationId -> location object so we don't need our own dependency to it.